deebot_client/commands/json/clean_logs_v2.py

```diff
@@ -0,0 +1,94 @@
+"""clean log commands."""
+import json
+from typing import Any
+
+from deebot_client.authentication import Authenticator
+from deebot_client.command import CommandResult
+from deebot_client.const import PATH_API_DLN_LOG_LIST, REQUEST_HEADERS
+from deebot_client.event_bus import EventBus
+from deebot_client.events import CleanJobStatus, CleanLogEntry, CleanLogEvent
+from deebot_client.logging_filter import get_logger
+from deebot_client.models import DeviceInfo
+
+from .common import JsonCommand
+
+_LOGGER = get_logger(__name__)
+
+
+class GetCleanLogsV2(JsonCommand):
+    """Get clean logs command."""
+
+    _targets_bot: bool = False
+    name = "clean"
+
+    def __init__(self, count: int = 0) -> None:
+        super().__init__({"count": count})
+
+    async def _execute_api_request(
+        self, authenticator: Authenticator, device_info: DeviceInfo
+    ) -> dict[str, Any]:
+        credentials = await authenticator.authenticate()
+
+        query_params = {
+            "did": device_info.did,
+            "res": device_info.resource,
+            "version": "v2",
+            "logType": self.name,
+            "channel": "google_play",
+            "size": "10",
+            "auth": json.dumps(
+                {
+                    "with": "users",
+                    "userid": credentials.user_id,
+                    "realm": "ecouser.net",
+                    "token": credentials.token,
+                    "resource": device_info.resource,
+                }
+            ),
```

Comment:
Are you sure that the auth must be passed as query param? Normally the authenticator will add it automatically but in the body...

Reply:
from what the app sends as request yes, this is the communication captured over mitm:

request:

```sh
GET https://api-app.dc-eu.ww.ecouser.net/app/dln/api/log/clean_result/list?did=REPLACED_DID&res=Gy2C&country=DE&size=10&auth=REPLACED_AUTH&version=v2&logType=clean&before=1699152310000&lang=EN&et1=1699152310545&defaultLang=zh_cn&channel=google_play HTTP/2.0
country: DE
accept: application/json
sign: REPLACED_SIGN
userid: REPLACED_USER_ID
token: REPLACED_TOKEN
authorization: Bearer REPLACED_TOKEN
v: 2.4.1
appid: ecovacs
plat: android
lang: EN
content-type: application/json
ts: 1699152310572
ucid: REPLACED_UC_ID
accept-encoding: gzip
user-agent: okhttp/4.9.1
content-length: 0
```

response:

```sh
HTTP/2.0 200
content-type: application/json; charset=utf-8
date: Sun, 05 Nov 2023 02:45:10 GMT
content-length: 40

{"code":0,"message":"success","data":[]}
```

request as curl:

```sh
curl -H 'country: DE' -H 'accept: application/json' -H 'sign: REPLACED_SIGN' -H 'userid: REPLACED_USER_ID' -H 'token: REPLACED_TOKEN' -H 'authorization: Bearer REPLACED_TOKEN' -H 'v: 2.4.1' -H 'appid: ecovacs' -H 'plat: android' -H 'lang: EN' -H 'content-type: application/json' -H 'ts: 1699152310572' -H 'ucid: REPLACED_UC:ID' --compressed -H 'user-agent: okhttp/4.9.1' 'https://api-app.dc-eu.ww.ecouser.net/app/dln/api/log/clean_result/list?did=REPLACED_DID&res=Gy2C&country=DE&size=10&auth=REPLACED_AUTH&version=v2&logType=clean&before=1699152310000&lang=EN&et1=1699152310545&defaultLang=zh_cn&channel=google_play'
```

Reply:
What happen if we omit the auth in the query params?
Does the request still work?